Src/FluentAssertions/Specialized/AsyncFunctionAssertions.cs

```diff
@@ -73,15 +73,12 @@ public void NotThrow(string because = "", params object[] becauseArgs)
             }
             catch (Exception exception)
             {
-                while (exception is AggregateException)
-                {
-                    exception = exception.InnerException;
-                }
+                var deepestException = GetDeepestException(exception);
 
                 Execute.Assertion
                     .BecauseOf(because, becauseArgs)
                     .FailWith("Did not expect any exception{reason}, but found a {0} with message {1}.",
-                        exception.GetType(), exception.Message);
+                        deepestException.GetType(), deepestException);
```

Comment:
❓ Can we safely assume the `ToString` of an exception always includes the `Message`?

Reply:
The default override of [`ToString()`](https://referencesource.microsoft.com/#mscorlib/system/badimageformatexception.cs,84ce150b21b0bc42,references) includes `Message` and I think we can rely on that.

For subclasses of `Exception` most seem not to override `ToString()`.

For subclasses overriding `Exception` I would rely on that they know what information is most relevant.
E.g. [`FileNotFoundException`](https://referencesource.microsoft.com/#mscorlib/system/io/filenotfoundexception.cs,e292457f6ecd914d,references).